app/src/main/java/com/github/orkest/View/MainActivity.kt

```diff
@@ -19,5 +29,61 @@ class MainActivity : AppCompatActivity() {
             }
             NavigationBar.CreateNavigationBar(navController = rememberNavController(), Constants.CURRENT_LOGGED_USER)
         }
+
+        createNotificationChannel(this)
+        promptUserToEnableNotifications(this)
+    }
+
+    /**
+     * The notification channel is required for Android 8.0 and above
+     * To inform the user to enable the notifications for this app
+     */
+    fun createNotificationChannel(context: Context) {
+        val name = "Your Channel Name"
+        val descriptionText = "Your Channel Description"
+        val importance = NotificationManager.IMPORTANCE_DEFAULT
+        val channel = NotificationChannel(NOTIFICATION_CHANNEL_ID, name, importance).apply {
+            description = descriptionText
+        }
+        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.createNotificationChannel(channel)
+    }
+
+    /**
+     * Build the notification of enabling them
+     */
+    fun promptUserToEnableNotifications(activity: Context) {
+        val notificationManager = NotificationManagerCompat.from(activity)
+        if (!notificationManager.areNotificationsEnabled()) {
+            val alertDialog = AlertDialog.Builder(activity)
+                .setTitle("Enable Notifications")
+                .setMessage("To receive important updates, please enable notifications for this app.")
+                .setPositiveButton("Enable", null)
+                .setNegativeButton("Cancel", null)
+                .create()
+
+            alertDialog.setOnShowListener {
+                alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).apply {
+                    setTextColor(Color.BLUE)
+                    setOnClickListener {
+                        val intent = Intent(Settings.ACTION_APP_NOTIFICATION_SETTINGS).apply {
+                            putExtra(Settings.EXTRA_APP_PACKAGE, activity.packageName)
+                        }
```

Comment:
Adding comments would be helpful to understand what the code does exactly

Reply:
Yep I've added some